pkg/server/server.go

```diff
@@ -151,7 +151,14 @@ func (s *Server) handle(apiOp *types.APIRequest, parser parse.Parser) {
 	} else if code > http.StatusOK {
 		apiOp.Response.WriteHeader(code)
 	}
-	metrics.RecordResponseTime(apiOp.Schema.ID, apiOp.Method, strconv.Itoa(code), float64(time.Since(requestStart).Milliseconds()))
+
+	var schemaID string
+	if apiOp.Schema != nil {
+		schemaID = apiOp.Schema.ID
+	} else {
+		schemaID = "unknown"
+	}
+	metrics.RecordResponseTime(schemaID, apiOp.Method, strconv.Itoa(code), float64(time.Since(requestStart).Milliseconds()))
 }
 
 func (s *Server) handleOp(apiOp *types.APIRequest) (int, interface{}, error) {
```

Comment:
Could you add a test to either here or rancher/rancher around it not panic'ing

Reply:
Added a test file for server.go. It's a bit of an odd test case because I'm testing that _nothing_ happens as opposed to checking that something happens or that I get a specific return value.

On the plus side it starts to add unit test coverage:

```
% go test -cover
PASS
	github.com/rancher/apiserver/pkg/server	coverage: 21.5% of statements
ok  	github.com/rancher/apiserver/pkg/server	0.328s
```